cmd/ttn-lw-cli/commands/end_devices.go

```diff
@@ -362,12 +362,7 @@ var (
 				}
 				paths = append(paths, ttnpb.FlattenPaths(decodedPaths, endDeviceFlattenPaths)...)
 
-				if ttnpb.ContainsField("supports_join", paths) {
-					if abp && device.SupportsJoin {
-						return errActivationMode.New()
-					}
-					abp = !device.SupportsJoin
-				}
+				abp = !device.SupportsJoin
```

Comment:
This now ignores the explicit values of the `--abp` and `--multicast` flags (L354-356) if we're receiving anything on stdin, while it previously only ignored those flags if the field `supports_join` was actually present in the input. Not sure if that's a big problem, but maybe good to warn if one of those flags is present while also reading from stdin.